### DISCUSSION TITLE
Storage policy with custom column

### DISCUSSION THREAD

**Discussion by bluebill1049** [supabase/supabase#2976] [Questions] [answered]
Hi there,

> This policy assumes that your table has a column "user_id", and allows users to delete rows which the "user_id" column matches their ID

<img width="940" alt="Screen Shot 2021-08-28 at 9 14 17 pm" src="https://user-images.githubusercontent.com/10513364/131216175-8fe18e1c-0830-4057-90b3-5eae987a4724.png">

If I want to create a policy against a bucket/object, how could I have a custom column such as `user_id`? Is this only applicable for tables instead of storages?

Thanks in advance.

**Comment by dshukertjr** [comment, marked as answer]
Hi @bluebill1049!

`objects` table in Supabase storage has a field named `owner`, which represents the user_id of the person who uploaded the objects. You could use this to implement a similar policy as the one you showed us!

Do you have a specific examples of what kind of policies you would like to set using custom column? If you have examples, we might be able to help you out!

**Reply by bluebill1049** [reply]
Thank you very much @dshukertjr for the answer. what's the best approach if I would like to associate the owner/user_id with an external table, such as `accounts` which have an associated `account_id`

Is the following a good approach:

```sql
(
  (bucket_id = 'name':: text)
  AND (
    (storage.foldername(name)) [ 1 ] = (
      SELECT
        (t.id):: text AS id
      FROM
        (
          users u
          JOIN accounts t ON ((t.id = u.account_id))
        )
      WHERE
        (uid() = u.id)
    )
  )
)
```

**Reply by dshukertjr** [reply]
It looks like you want to store the file under `[account_id]/`. In that case, would something like this work?
```sql
bucket_id = 'name':: text
  and
storage.foldername(name)[1] = (select account_id from users where id = uid())
  and
uid() = owner
```

**Reply by bluebill1049** [reply]
This is awesome! thanks a lot, this is a lot cleaner and efficient.
